lib/gst/player/gstplayer.c

```diff
@@ -688,6 +689,44 @@ eos_cb (GstBus * bus, GstMessage * msg, gpointer user_data)
   self->priv->is_eos = TRUE;
 }
 
+static gboolean gst_player_set_playback_rate_internal (gpointer user_data)
+{
+  GstPlayer *self = GST_PLAYER (user_data);
+  GstStateChangeReturn state_ret;
+  GstClockTime pos;
+
+  GST_DEBUG_OBJECT (self, "Play");
+  if (self->priv->current_state < GST_STATE_PAUSED) {
+    return;
+  }
+  g_mutex_lock (&self->priv->lock);
+  if (!self->priv->uri) {
+    g_mutex_unlock (&self->priv->lock);
+    return G_SOURCE_REMOVE;
+  }
+  g_mutex_unlock (&self->priv->lock);
+  pos = gst_player_get_position (self);
+  gst_element_seek (self->priv->playbin,
+                    self->priv->rate,
+                    GST_FORMAT_TIME,
+                    GST_SEEK_FLAG_FLUSH | GST_SEEK_FLAG_ACCURATE,
+                    GST_SEEK_TYPE_SET,
+                    pos,
+                    GST_SEEK_TYPE_NONE,
+                    0);
+  
+  gst_element_set_state (self->priv->playbin, GST_STATE_PLAYING);
+
+  return G_SOURCE_REMOVE;  
+}
+
+void gst_player_set_playback_rate (GstPlayer * self, gdouble rate)
+{
+  g_return_if_fail (GST_IS_PLAYER (self));
+  self->priv->rate = rate;
```

Comment:
Normal seek events also have to take the rate into account. I think it would make sense to just update the rate, and the reuse the existing seeking logic (plus adaptions) instead of writing another completely separate seek function.